sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -177,18 +177,22 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
             }
 
             return this.shouldRetryOnRequestTimeout(
+                clientException,
                 this.isReadRequest,
                 this.request.getNonIdempotentWriteRetriesEnabled(),
                 clientException.getSubStatusCode());
         }
 
-        if (clientException != null && Exceptions.isStatusCode(clientException, HttpConstants.StatusCodes.INTERNAL_SERVER_ERROR)) {
+
+        if (clientException != null
+            && Exceptions.isStatusCode(clientException, HttpConstants.StatusCodes.INTERNAL_SERVER_ERROR)
+            && !Exceptions.isClientAssignedSubStatusCodeForInternalServerError(clientException.getStatusCode(), clientException.getSubStatusCode())) {
 
             if (logger.isDebugEnabled()) {
                 logger.info("Internal server error - IsReadRequest {}", this.isReadRequest, e);
```

Comment:
can we please fix this logging? its getting logged as info on debug enabled.

Reply:
Fixed